### DISCUSSION TITLE
Problems with bmodes input files

### DISCUSSION THREAD

**Discussion by tatesnow** [OpenFAST/openfast#947] [Q&A]
Hi @jjonkman, 
I recently wanted to modify the parameters of a Bladed software into Bmodes input file format to enable modal and frequency calculations via Bmodes. I can currently obtain the center of mass and mass of the wind wheel and nacelle through the Bladed software. How do I calculate the mass moment of inertia of tower?
`  ixx_tip     blade lag or tower s-s mass moment of inertia about the tip-section x reference axis (kg-m^2)
 iyy_tip     blade flap or tower f-a mass moment of inertia about the tip-section y reference axis (kg-m^2)
 izz_tip     torsion mass moment of inertia about the tip-section z reference axis (kg-m^2)
 ixy_tip     cross product of inertia about x and y reference axes(kg-m^2)
 izx_tip     cross product of inertia about z and x reference axes(kg-m^2)
iyz_tip     cross product of inertia about y and z reference axes(kg-m^2)`
Best regard！

**Comment by jjonkman** [comment]
Dear @tatesnow,

These BModes inputs are referring to the second mass moments of inertia of the rotor-nacelle assembly (RNA).  Other than a laborious hand calculation, we've used two methods to derive these, e.g., by calculating the 6x6 rigid-body mass matrix of the RNA through a FAST / OpenFAST linearization process or using a CAD-based preprocessor to derive the aggregate mass and inertia.

Best regards,
